src/generators/Silk.NET.SilkTouch.Symbols/StructSymbol.cs

```diff
@@ -9,11 +9,31 @@ namespace Silk.NET.SilkTouch.Symbols;
 /// A <see cref="TypeSymbol"/> representing a <c>struct</c>.
 /// </summary>
 /// <param name="Identifier">The Identifier of this struct</param>
-/// <param name="Members">The Members of this struct</param>
+/// <param name="Layout">The layout of this struct</param>
 /// <remarks>
 /// In this context, a Struct means a type that represents the layout of a continuous block of memory.
 /// </remarks>
 // /// Each meaningful place in this memory called a field (see <see cref="FieldSymbol"/>) is accessible via this type.
 // /// Fields are allowed to overlap.
 // /// Additionally it may contain one or multiple <see cref="MethodSymbol"/> that are called with an instance of this type as their first argument.
-public sealed record StructSymbol(IdentifierSymbol Identifier, ImmutableArray<MemberSymbol> Members) : TypeSymbol(Identifier);
+public sealed record StructSymbol(IdentifierSymbol Identifier, StructLayout Layout) : TypeSymbol(Identifier);
+
+/// <summary>
+/// A <see cref="StructSymbol"/> representing the layout of a <see cref="StructSymbol"/>
+/// </summary>
+/// <param name="Entries">The entries of this layout</param>
+public sealed record StructLayout(ImmutableArray<LayoutEntry> Entries)
+{
+    /// <summary>
+    /// An empty layout with no members
+    /// </summary>
+    public static readonly StructLayout Empty = new StructLayout(ImmutableArray<LayoutEntry>.Empty);
+}
+
+/// <summary>
+/// Represents an entry in a <see cref="StructLayout"/>
+/// </summary>
+/// <param name="Member">The member symbol associated with this layout entry</param>
+/// <param name="ByteOffset">The offset of this entry in bytes</param>
+/// <seealso cref="StructLayout"/>
+public sealed record LayoutEntry(MemberSymbol Member, int ByteOffset);
```

Comment:
Byte offsets can be different on 64-bit and 32-bit. e.g.:

```cs
public struct X
{
    [FieldOffset(0)]
    public nint Y;
    [FieldOffset(sizeof(nint))]
    public nint Z;
}
```

How would we represent this?

Reply:
Not sure, but I'm not sure what the expected emitted code would be either, sizeof(nint) is not constant.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGAOgCUBXAOwwEt8YaBJFmKCAA4BlPgDdWYGLgDcAWABQAbSEYoDMBgAy2AJ4QGGABTa9BgNKsmAExoBRBAIA2E1hgCUAXXILiAZnK4quoY5AAaCgDeCuQx5IoAYqwwjlYA8gBm6bgwRqSe0bF+5EyWIQCacvKxcYnJaZnZRrisAF4wEOmGJSxu+VWF/t0hAFqVAL5AA==